examples/ffd/bsp_config/XK_VOICE_L71/platform/driver_instances.h

```diff
@@ -55,4 +62,17 @@ extern rtos_i2c_master_t *i2c_master_ctx;
 extern rtos_i2s_t *i2s_ctx;
 extern rtos_uart_tx_t *uart_tx_ctx;
 
+#if appconfRECOVER_MCLK_I2S_APP_PLL
+typedef struct {
+    port_t p_mclk_count;                    // Used for keeping track of MCLK output for sw_pll
+    port_t p_bclk_count;                    // Used for keeping track of BCLK input for sw_pll
+    sw_pll_state_t *sw_pll;                 // Pointer to sw_pll state (if used)
+
+}sw_pll_ctx_t;
+
+static sw_pll_state_t sw_pll = {0};
```

Comment:
It seems odd to have two copies of the SW PLL state, one in sw_pll and another as an element in *sw_pll_ctx.  Any idea why that has been done?

Reply:
I don't understand this comment. In sw_pll_ctx, we have a pointer to sw_pll_state_t. This pointer is assigned to the static struct in [sw_pll_platform_init(](https://github.com/lucianomartin/sln_voice/blob/feature/ffd_i2s/examples/ffd/bsp_config/XK_VOICE_L71/platform/platform_init.c#L150)). I see one copy in the line pointed by this comment, but I can't find the other copy.

Reply:
Sorry, you're right.  I didn't notice that the `sw_pll` element in `sw_pll_ctx_t` points to the `sw_pll` static variable defined on line 73.  Please ignore my original comment.